File: backend/customer-service/src/main/java/com/loanflow/customer/service/EkycService.java
package com.loanflow.customer.service;

import com.loanflow.dto.response.EkycInitiateResponse;
import com.loanflow.dto.response.EkycVerifyResponse;
import com.loanflow.dto.response.KycStatusResponse;

import java.util.UUID;

/**
 * Service interface for UIDAI Aadhaar e-KYC verification (US-029).
 * Implementations: MockEkycService (dev/test), UidaiEkycService (uat/prod).
 */
public interface EkycService {

    /**
     * Initiate OTP-based e-KYC verification by sending OTP to Aadhaar-linked mobile.
     *
     * @param customerId   the customer UUID
     * @param aadhaarNumber 12-digit Aadhaar number
     * @return initiation response with transaction ID and masked mobile
     */
    EkycInitiateResponse initiateOtp(UUID customerId, String aadhaarNumber);

    /**
     * Verify OTP and retrieve e-KYC demographic data from UIDAI.
     *
     * @param customerId    the customer UUID
     * @param transactionId the transaction ID from initiation
     * @param otp           the 6-digit OTP entered by the user
     * @return verification response with e-KYC data if successful
     */
    EkycVerifyResponse verifyOtp(UUID customerId, String transactionId, String otp);

    /**
     * Get current KYC verification status for a customer.
     *
     * @param customerId the customer UUID
     * @return KYC status response
     */
    KycStatusResponse getKycStatus(UUID customerId);
}

File: frontend/loanflow-web/src/app/features/customer/services/ekyc.service.ts
import { Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { Observable } from 'rxjs';
import { ApiResponse } from '../models/customer.model';
import {
  EkycInitiateResponse,
  EkycVerifyRequest,
  EkycVerifyResponse,
  KycStatusResponse
} from '../models/ekyc.model';

/**
 * Angular service for e-KYC verification API (US-029).
 */
@Injectable({ providedIn: 'root' })
export class EkycService {

  private apiUrl = '/api/v1/customers';

  constructor(private http: HttpClient) {}

  /**
   * Initiate e-KYC OTP verification for a customer.
   */
  initiateEkyc(customerId: string, aadhaarNumber: string): Observable<ApiResponse<EkycInitiateResponse>> {
    return this.http.post<ApiResponse<EkycInitiateResponse>>(
      `${this.apiUrl}/${customerId}/ekyc/initiate`,
      { aadhaarNumber }
    );
  }

  /**
   * Verify OTP and retrieve e-KYC demographic data.
   */
  verifyOtp(customerId: string, request: EkycVerifyRequest): Observable<ApiResponse<EkycVerifyResponse>> {
    return this.http.post<ApiResponse<EkycVerifyResponse>>(
      `${this.apiUrl}/${customerId}/ekyc/verify`,
      request
    );
  }

  /**
   * Get current KYC verification status for a customer.
   */
  getKycStatus(customerId: string): Observable<ApiResponse<KycStatusResponse>> {
    return this.http.get<ApiResponse<KycStatusResponse>>(
      `${this.apiUrl}/${customerId}/ekyc/status`
    );
  }
}
